File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0 for _ in range(len(prices))]
    stack = []
    
    for i in range(len(prices) - 1):
        cur_price = prices[i]        
        for j in range(i, len(prices) - 1):
            if cur_price <= prices[j]:
                stack.append(prices[j])
            else:
                break
            
        # print(stack)
        answer[i] = len(stack)
        
        stack = []
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int N = prices.length;
        int[] answer = new int[N];
        Stack<Integer> stack = new Stack<>();        
        stack.add(0);
        
        for(int i = 1; i < N; i++) {
            int curPrice = prices[i];
            // System.out.println(stack + ", " + stack.peek()  + ", " +  curPrice);            
            while(!stack.isEmpty() && prices[stack.peek()] > curPrice) {
                int curIdx = stack.pop();                    
                answer[curIdx] = i - curIdx;                
            }
            stack.add(i);
            // System.out.println(Arrays.toString(answer));
        }
        // System.out.println(stack + ", " + stack.peek());
        while(!stack.isEmpty()) {
            int curIdx = stack.pop();                    
            answer[curIdx] = N - curIdx - 1;
        }
        
        return answer;
    }
}